src/export/StructureDefinitionExporter.ts

```diff
@@ -61,16 +78,244 @@ export class StructureDefinitionExporter implements Fishable {
     private readonly fisher: MasterFisher
   ) {}
 
+  /**
+   * Processes the fshDefinition's parent, validating it according to its type.
+   * Returns the parent's StructureDefinition as the basis for the new StructureDefinition
+   * for the provided fshDefinition.
+   * @param {Extension | Profile | Logical | Resource} fshDefinition - The definition
+   *        to be preprocessed. It is updated directly based on this processing.
+   * @returns {StructureDefinition} for this fshDefinition
+   * @private
+   */
+  private getStructureDefinition(
+    fshDefinition: Profile | Extension | Logical | Resource
+  ): StructureDefinition {
+    // Process/validate the fshDefinition.parent value with the purpose of
+    // obtaining the parent's StructureDefinition as the basis for this
+    // fshDefinition's StructureDefinition.
+    // RECALL: fshDefinition.parent can be specified as the 'id', 'name', or 'url'.
+
+    if (isEmpty(fshDefinition.parent)) {
+      // Handle cases where the parent is not specified by throwing an error.
+      // - Profile has a hard requirement to define a parent and does not have a default
+      //   parent defined in the Profile class. Therefore, we need to throw this error.
+      // - Extension, Logical, and Resource all have a default parent, so we should never
+      //   get to this point where it is not defined. That said, if something somewhere
+      //   in the code were to change/remove the default, this error will then be thrown.
+      throw new ParentNotProvidedError(fshDefinition.name, fshDefinition.sourceInfo);
+    }
+
+    let defnType: Type;
+    if (fshDefinition instanceof Extension) {
+      defnType = Type.Extension;
+    } else if (fshDefinition instanceof Logical) {
+      defnType = Type.Logical;
+    } else {
+      defnType = Type.Resource;
+    }
```

Comment:
Good points! I really did not think this through. I'll fix this as suggested.

Reply:
TODO: PAQ - Resolve fishForMetadata issue:
Searches for the Metadata associated with the passed in name/id/url.  It will first search through the local package (which contains FHIR artifacts exported so far), then through the tank, then through the external FHIR definitions.
This means that when fishing for metadata to get the "result?.url for throwing the `ParentDeclaredAsNameError` and `ParentDeclaredAsIdError`, it will get the fshDefinition's URL rather than the parent's.